src/Mod/Draft/DraftGeomUtils.py

```diff
@@ -47,7 +47,14 @@
 
 def precision():
     "precision(): returns the Draft precision setting"
-    return params.GetInt("precision",6)
+    # Set precision level with a cap to avoid overspecification that:-
+    #  1 - whilst it is precise enough (e.g. that OCC would consider 2 points are conincident) (not sure what it should be 10 or otherwise);
+    #  2 - but FreeCAD / OCC can handle 'internally' (e.g. otherwise user may set something like 15 that the code would never consider 2 points are coincident as internal float is not that precise);
```

Comment:
I wonder if we shouldn't make the user aware of this? For ex. make the preference setting widget unable to go above 10

Reply:
Thanks.  I have thought about that but without overall understanding of FC / OCC underlying mechanism, and not checking any code about preference, I have not much idea how this should be handled.

First, not sure if this precision level is relative if difference units system is used (e.g. mm vs inches) and if that will create other problem.

Second, I am just guessing it was OCC that determine if 2 endpoints are 'connected' / 'within tolerance', but if sketch is of same mechanism....

Finally, a figure of 10 for precision is my understanding in trial and error checking the figures in report view.

See what else I can help.

Reply:
Or after all, in fact, this should not be set by user ... ?

... as on one hand, this should be precise enough so e.g. 2 endpoints recognised at a precision level in a script / method / function as Coincidence should be same as OCC recognised...

... but on the other, if this is 'too precise', e.g. 2 endpoints 'thought' not coincident at the user's set precision level in a script / method / function, should have been recognised as coincident in OCC (or beyond FC / OCC 'internal' mechanism) [ my previous error ]

... so the question is, is there a range between the above 2 available?

I am not sure if my understanding is correct :)

Reply:
Still not very happy about limiting the precision value without the user knowing it... I'll merge this PR and alter the preferences dialog so the precision value cannot go above 10

Reply:
No idea about the 'maximum' also.  Seem in Axis discussion below, 9 or below or it would find all intersections :)

[https://forum.freecadweb.org/viewtopic.php?f=23&t=35300](url)

Reply:
Okay... Let's already merge this one, and keep investigations going.